dask_jobqueue/deploy/cluster_manager.py

```diff
@@ -83,8 +84,14 @@ def __init__(self, adaptive_options={}):
         self._adaptive_options = adaptive_options
         self._adaptive_options.setdefault('worker_key', self.worker_key)
 
+    def wait_until_n_workers(self, n):
+        '''Block by sleeping until we have n active workers'''
+        while n and len(self.scheduler.workers) < n:
+            time.sleep(1)
```

Comment:
If we keep this function then I would decrease this sleep significantly, probably to around 10ms.  It'll still be way less than 1% cpu time.

It's common in Dask to build functions that are async-friendly first, and then wrap around them with `self.cluster.sync`.  If we're supporting Python 2 then we do this with `@gen.coroutine`, if we're not then we use `async def`.